### ISSUE TITLE
App Crash after upgrade to V3.0

### ISSUE THREAD

**Issue by eeease** [RenderHeads/UnityPlugin-AVProVideo#1811]
### Please DO NOT LINK / ATTACH YOUR PROJECT FILES HERE

**Describe the issue**
After updating to AVPro 3.0 Ultra, playback in our app causes the entire app to crash.
We specifically upgraded to 3.0 in order to use Vulkan, but now our playback crashes the app in both Vulkan and OpenGLES3.
 
**Your Setup (please complete the following information):**
 - Unity version: 2022.3.4f1
 - AVPro Video version (number and edition (trial/core/ultra/enterprise)): 3.0Ultra
 - Operating system version: Android 13 (API Level 33)
 - Device model: Meta Quest 2
 - Video specs (resolution, frame-rate, codec, file size): Multiple videos, including:
     - the BigBuckBunny stream located in your FAQ.
     - An .mp4 dimensions: 2500x1248; H264
     - An .mp4 dimensions: 2880x2880; H265
 - Color Space: Linear
 - Oculus Stereo Rendering Mode: Multi Pass (also crashes in Multiview).
 - ExoPlayer > Platform Specific > Android > Texture Format: Y Cb Cr 420_OES (also crashes with BGRA) 

**Logs**
If applicable, add error logs to help explain your problem.
[logcat_multipass_vulkan_loadNeonPink.txt](https://github.com/RenderHeads/UnityPlugin-AVProVideo/files/14890450/logcat_multipass_vulkan_loadNeonPink.txt)

### Please DO NOT LINK / ATTACH YOUR PROJECT FILES HERE
Instead email the link to us unitysupport@renderheads.com

Any suggestions would be very helpful.  I've been nonstop testing different variables for about a week.
Thank you!

**Comment by eeease** [comment]
Runs fine in Editor, by the way.  No problems loading any of the listed video files.
Standalone Windows build also runs fine, plays back video without issue.

**Comment by Chris-RH** [comment]
Did you follow our [upgrade steps](https://www.renderheads.com/content/docs/AVProVideo-v3/articles/upgrading-from-avpro-video-2x.html)?
Can you reproduce this in a new project using only AVPro Video mediaplayer demo?

**Comment by eeease** [comment]
Yep, I followed the upgrade steps, thanks.

When I export a build with just AVPro Video Mediaplayer Demo, the mediaplayer opens on launch and plays through the video.
Vulkan graphics API, Android 13.
I tried with the Exo Player, BGRA and OES texture format options.

**Comment by Anton111111** [comment]
> Yep, I followed the upgrade steps, thanks.


I have crash after upgrade too. But it doesn't work with OpenGLES3 but works with Vulkan ;) https://github.com/RenderHeads/UnityPlugin-AVProVideo/issues/1807

Did you try build Developer release? I have not crash in Developer release.

**Comment by Hadrien-Estela** [comment]
Experiencing same issue.
I tried with vulkan but the quest app does not start if in build in dev mode.

https://issuetracker.unity3d.com/issues/quest-openxr-vulkan-application-stuck-on-loading-when-launching-on-meta-quest-2-slash-3-headset-with-development-build-enabled

**Comment by eeease** [comment]
> > Yep, I followed the upgrade steps, thanks.
> 
> I have crash after upgrade too. But it doesn't work with OpenGLES3 but works with Vulkan ;) #1807
> 
> Did you try build Developer release? I have not crash in Developer release.

I saw your ticket, yeah!  Weird that mine broke with the switch to Vulkan and yours broke with the switch to OpenGLES3!

I have been making _mostly_ development builds.  However, I've tried release builds, too and they also crash when I load a video.

I've also tried having it connected or disconnected to my computer while running the app, since I read somewhere that being connected to logcat and/or Unity could cause some weird bugs.

**Comment by eeease** [comment]
@Chris-RH 
Any chance the crash has to do with line 803:
`java_vm_ext.cc:579] JNI DETECTED ERROR IN APPLICATION: JNI CallStaticVoidMethodV called with pending exception java.lang.IllegalArgumentException:`

We're using
Unity 2022.3.4f1 with Android tools that are "installed with Unity."
It's not easy to parse exactly what JDK tools we're using but a notice in Preferences says "based on OpenJDK 11.0.14.1+1."

Oh, also, my Quest 2 device is on Software version 63.  Apparently 64 just released this week but I'm not getting an option to update my device yet.

**Comment by Chris-RH** [comment]
Out of interest, have you tested this on any other android device?

**Comment by Chris-RH** [comment]
Which OS version are you using on the Quest?

**Comment by eeease** [comment]
> Have you tried a vulkan build without AVPro in the project?

Yes.  It runs okay without AVPro.

I'm seeing a common link between all of my crashes - they all mention AVPro_Seek:
![image](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/10697313/15df7e83-e3de-4dd9-9712-b8023ed117cf)

Is there anything new in Version 3.0 that's happening with Seek that didn't happen in 2.x?
I'm going to try to make a build with Seek disabled.

**Comment by Last8Exile** [comment]
On android application crashes on MediaPlayer.Control.SeekFast(). Crash does not happen on Seek().
I tested different cases and values. It always crashes with SeekFast().

Unity 2022.3.22f1
Devlopment Build
AVPro v3.0.0f1-ultra
armV7
Quest 2
Min SDK 29
Target SDK 32
Compile SDK 33

```
JNI DETECTED ERROR IN APPLICATION: JNI CallStaticVoidMethodV called with pending exception java.lang.IllegalArgumentException:
  at void androidx.media3.common.util.Assertions.checkArgument(boolean) (Assertions.java:40)
  at void androidx.media3.exoplayer.SeekParameters.<init>(long, long) (SeekParameters.java:71)
  at void com.renderheads.AVPro.Video.Player_ExoPlayer._seekWithTolerances(long, long, long) (SourceFile:-1)
  at void com.renderheads.AVPro.Video.Manager.Player_Seek(int, double, double, double) (SourceFile:-1)
  at boolean com.unity3d.player.UnityPlayer.nativeRender() ((null):-2)
  at boolean com.unity3d.player.UnityPlayer.-$$Nest$mnativeRender(com.unity3d.player.UnityPlayer) ((null):-1)
  at boolean com.unity3d.player.UnityPlayer$F$a.handleMessage(android.os.Message) ((null):-1)
  at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
  at boolean android.os.Looper.loopOnce(android.os.Looper, long, int) (Looper.java:214)
  at void android.os.Looper.loop() (Looper.java:304)
  at void com.unity3d.player.UnityPlayer$F.run() ((null):-1)

    in call to CallStaticVoidMethodV
    from boolean com.unity3d.player.UnityPlayer.nativeRender()
```

```
managed backtrace:
      #00 (wrapper managed-to-native) RenderHeads.Media.AVProVideo.PlatformMediaPlayer/Native:AVPPlayerUpdate (intptr)
      #01 RenderHeads.Media.AVProVideo.PlatformMediaPlayer:Update ()
      #02 RenderHeads.Media.AVProVideo.PlatformMediaPlayer:SeekWithTolerance (double,double,double) 
      #03 RenderHeads.Media.AVProVideo.PlatformMediaPlayer:SeekFast (double) 
      #04 VideoUtils.VideoPlayerImpl/<SeekAsync>d__58:MoveNext () // _player.Control.SeekFast(0);
```

**Comment by eeease** [comment]
Instead of upgrading from 2.x to 3.0, I just tried deleting the previous package completely then re-downloading and re-importing 3.0 ultra.

The same crash occurs upon playback - content plays for a second then the app crashes.  OpenGLES3, Quest 2, Linear Color Space, ExoPlayer (BGRA).

tombstone file attached as txt file.

[tombstone_newAVPro30Install.txt](https://github.com/RenderHeads/UnityPlugin-AVProVideo/files/14951168/tombstone_newAVPro30Install.txt)

**Comment by Chris-RH** [comment]
How does it work if you use Android Mediaplayer instead of Exoplayer?

**Comment by Last8Exile** [comment]
I think issue with AVPro calls `void androidx.media3.exoplayer.SeekParameters` with invalid arguments.
MediaPlayer.Control.SeekFast() on C# side passes `double.PositiveInfinity` and `double.NegativeInfinity` to SeekWithTolerance.
Seems they are valid with exoplayer and invalid with media3.
And unhandled exception kills application.

**Comment by Anton111111** [comment]
> I think issue with AVPro calls `void androidx.media3.exoplayer.SeekParameters` with invalid arguments.

In my case app crash immediatly on start app. But on the first scene i don't have AvPro at all.

**Comment by eeease** [comment]
> Just rereading this thread. You can't use OES rendering with Vulkan, it is OpenGLES and Exoplayer only. Probably not relevant, because you said it crashed in BGRA as well.
> 
> Having a bit of a google, there do seem to be quite a few threads popping up regarding Vulkan and memory issues. Are you using SRP, URP or HDRP?
> 
> It will be interesting to see if there is any improvement after updating to Meta's v64

Thanks for the info.
We are using URP at the moment.  No discrete plans to change to HDRP, though it's not out of the realm of possibility.

**Comment by Anton111111** [comment]
> It will be interesting to see if there is any improvement after updating to Meta's v64

In my case without changes after upgrade to v64

**Comment by Chris-RH** [comment]
@eeease 
Does this still occur when using SRP?

**Comment by eeease** [comment]
> @eeease Does this still occur when using SRP? What URP setting is it? High Fidelity/Balanced/Performant? Have you tried the other ones?

I'm not really sure how to answer those questions.
We are using a Universal Render Pipeline asset that is assigned to Graphics > Scriptable Render Pipeline Settings.  Does that mean we're using SRP, but with a URP Asset?

I don't see High Fidelity/Balanced/Performant as options anywhere.

I'm attaching screenshots of our Graphics Assets, including the Inspector.

![URP1](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/10697313/bdf6df3c-08d8-4767-84d2-cfc065bdf0c9)
![URP3](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/10697313/65d74ee3-8f4e-4043-b7ea-67cab896c621)
![URP2](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/10697313/ad4e014f-106c-4868-a41b-32f62660460d)

**Comment by Chris-RH** [comment]
@Last8Exile thanks, I have passed that information on to the dev team

@eeease They are in the Quality settings section.  I think it defaults to High Fidelity.  I don't have a Quest 2 to hand, so I'm interested to know if changing the settings helps at all.

![quality](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/102963025/c8ca88c9-843d-494b-aa9a-2e04ab7d54d0)

**Comment by eeease** [comment]
@Chris-RH We set Android to default to Very Low, but something that's interesting is that Quest - Android is its own Level, not a platform...?
![image](https://github.com/RenderHeads/UnityPlugin-AVProVideo/assets/10697313/5d585138-1c14-42c6-b2e3-783438fc198a)

**Comment by Chris-RH** [comment]
Ahh, ok, thanks :) 

I'll update you when the dev team has managed to look into the Seek issue.

**Comment by SlimsDonxi** [comment]
> > I think issue with AVPro calls `void androidx.media3.exoplayer.SeekParameters` with invalid arguments.
> 
> In my case app crash immediatly on start app. But on the first scene I don't have AvPro at all.

Same here, brand new project, just imported the package !

App crashes instantly on press, before the Unity logo
Abort message: 'JNI DETECTED ERROR IN APPLICATION: mid == null
    in call to CallStaticObjectMethodV'

All arm64 plugins are correctly set to Android CPU: ARM64
Clearly the libAVProVideo2Native.so is causing all the issues, but I got no clue how to go from here !

**Comment by WikkidEdd** [comment]
We're having the same issue as the OP i.e the Seek causing a crash due to invalid arguments. 

@Last8Exile's summary of the problem is spot on. You can work around the issue by changing this line  `SeekWithTolerance(toTime, double.PositiveInfinity, double.PositiveInfinity);` in Assets\Plugins\AVProVideo\Runtime\Scripts\Internal\Players\PlatformMediaPlayer.cs to something more reasonable that won't trigger the overflow.

We've set it to `SeekWithTolerance(toTime, 9999, 9999);`

**Comment by Ste-RH** [comment]
RE Seek issue.  You are 100% correct.  Thanks for digging into it.  This will be fixed in v3.0.2 (penciled for release this week)

**Comment by Chris-RH** [comment]
AVPro Video version 3.0.2 has been released.  Please let us know if it has not fixed your issue.
